Tbh I'm not entirely sure why this change causes new type errors in `util/unit_system.py`. Will investigate it.
```py
Error: homeassistant/util/unit_system.py:44: error: Cannot determine type of "VALID_UNITS"  [has-type]
Error: homeassistant/util/unit_system.py:53: error: Cannot determine type of "VALID_UNITS"  [has-type]
Error: homeassistant/util/unit_system.py:55: error: Cannot determine type of "VALID_UNITS"  [has-type]
Error: homeassistant/util/unit_system.py:57: error: Cannot determine type of "VALID_UNITS"  [has-type]
```

--- a/homeassistant/util/signal_type.py
+++ b/homeassistant/util/signal_type.py
@@ -2,8 +2,6 @@
 
 from __future__ import annotations
 
-from typing import Any
-
 
 class _SignalTypeBase[*_Ts](str):
     """Generic base class for SignalType."""
@@ -21,7 +19,3 @@ class SignalTypeFormat[*_Ts](_SignalTypeBase[*_Ts]):
     """Generic string class for signal. Requires call to 'format' before use."""
 
     __slots__ = ()
-
-    def format(self, *args: Any, **kwargs: Any) -> SignalType[*_Ts]:
-        """Format name and return new SignalType instance."""
-        return SignalType(super().format(*args, **kwargs))

--- a/homeassistant/util/signal_type.pyi
+++ b/homeassistant/util/signal_type.pyi
@@ -31,6 +31,7 @@ def _test_signal_type_typing() -> None:  # noqa: PYI048
 
     This is tested during the mypy run. Do not move it to 'tests'!
     """
+    # pylint: disable=import-outside-toplevel
     from homeassistant.core import HomeAssistant
     from homeassistant.helpers.dispatcher import (
         async_dispatcher_connect,

--- a/homeassistant/util/signal_type.pyi
+++ b/homeassistant/util/signal_type.pyi
@@ -3,12 +3,6 @@
 
 from typing import Any, assert_type
 
-from homeassistant.core import HomeAssistant
-from homeassistant.helpers.dispatcher import (
-    async_dispatcher_connect,
-    async_dispatcher_send,
-)
-
 __all__ = [
     "SignalType",
     "SignalTypeFormat",
@@ -37,6 +31,12 @@ def _test_signal_type_typing() -> None:  # noqa: PYI048
 
     This is tested during the mypy run. Do not move it to 'tests'!
     """
+    from homeassistant.core import HomeAssistant
+    from homeassistant.helpers.dispatcher import (
+        async_dispatcher_connect,
+        async_dispatcher_send,
+    )
+
     hass: HomeAssistant
     def test_func(a: int) -> None: ...
     def test_func_other(a: int, b: str) -> None: ...

